packages/smb/src/components/ui/offices-tab/offices-tab.tsx

```diff
@@ -0,0 +1,137 @@
+  pageNumber?: number
+  pageSize?: number
+  totalCount?: number
+  dataTable: DataTableRow[][]
+  handleChangePage: (page: number) => void
+}
+
+export const renderContent = ({
+  loading,
+  error,
+  dataTable,
+  pageNumber = 0,
+  pageSize = 0,
+  totalCount = 0,
+  handleChangePage,
+}: RenderContentParams) => {
+  if (loading) {
+    return <Loader />
+  }
+  if (error) {
+    return <Alert message={error.message} type="danger" />
+  }
+  return (
+    <React.Fragment>
+      <Spreadsheet
+        data={dataTable as Cell[][]}
+        description={
+          <p>
+            <p>
+              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
+              industry&apos;s standard dummy text ever since the 1500s, when an unknown printer took a galley of type
+              and scrambled it to make a type specimen book.
+            </p>
+          </p>
+        }
+      />
+      <Section>
+        <Pagination pageNumber={pageNumber} pageSize={pageSize} totalCount={totalCount} onChange={handleChangePage} />
+      </Section>
+    </React.Fragment>
+  )
+}
+
+export const handleChangePage = ({ history }) => (pageNumber: number) => {
+  const searchParams = queryString.stringify({
+    page: pageNumber,
+  })
+  history.push({
+    search: searchParams,
+  })
+}
+
+export const OfficesTab: React.FC<OfficesTabProps> = ({ location, history }) => {
+  const params = queryString.parse(location?.search)
+  const page = Number(params?.page) || 1
+  const { loading, error, data } = useQuery<OfficesQueryResponse, Offices>(OFFICES, {
+    variables: { PageSize: OFFICES_PER_PAGE, PageNumber: page },
+    fetchPolicy: 'network-only',
+  }) as QueryResult<OfficesQueryResponse, Offices>
+
+  const dataTable = getDataTable(data || { GetOffices: { _embedded: [] } })
+
+  return (
+    <div>
+      {renderContent({
+        loading,
+        error,
+        dataTable,
+        pageNumber: data?.GetOffices?.pageNumber,
+        pageSize: data?.GetOffices?.pageSize,
+        totalCount: data?.GetOffices?.totalCount,
+        handleChangePage: handleChangePage({ history }),
+      })}
+    </div>
+  )
+}
+
+export function getDataTable(data: OfficesQueryResponse): DataTableRow[][] {
+  let dataTable: DataTableRow[][] = [tableHeaders]
+  const offices: OfficeModel[] = data.GetOffices?._embedded || []
+  const dataRows: DataTableRow[][] = offices.map((office: OfficeModel) => [
+    { value: office.name },
+    { value: office.address?.buildingName },
+    { value: office.address?.buildingNumber },
+    { value: office.address?.line1 },
+    { value: office.address?.line2 },
+    { value: office.address?.line3 },
+    { value: office.address?.line4 },
+    { value: office.address?.postcode },
+    { value: office.workPhone },
+    { value: office.email },
+    { value: office.email },
+  ])
+  dataTable = [tableHeaders, ...dataRows]
+  return dataTable
+}
+
+export default withRouter(OfficesTab)
```

Comment:
Its cleaner and simpler if we can use new react router hooks instead of withRouter HOC
useParams: https://reacttraining.com/blog/react-router-v5-1/#useparams
useHistory: https://reacttraining.com/blog/react-router-v5-1/#usehistory

Reply:
> Its cleaner and simpler if we can use new react router hooks instead of withRouter HOC
> useParams: https://reacttraining.com/blog/react-router-v5-1/#useparams
> useLocation: https://reacttraining.com/blog/react-router-v5-1/#uselocation
> useHistory: https://reacttraining.com/blog/react-router-v5-1/#usehistory

@nphivu414 I get your point but I think we should use the old style coding for consistence with old code.

Reply:
@trankhacvy for the new smb app we can apply th enew things, so I think if we can change pls go ahead